src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReferenceHandlingStrategy.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization
+{
+    internal enum ReferenceHandlingStrategy
```

Comment:
For perf reasons, I added the field ReferenceHandlerStrategy directly to JsonSerializerOptions, 

Originally I was using a bool inside ReferenceHandler but the property check and null check (`if (options.ReferenceHandler != null && !options.ReferenceHandler.UsePreserveSemantics`) was adding a significant overhead to the benchmarks, that's why I added the field in options, at the same level of ReferenceHandler.

**https://github.com/dotnet/runtime/blob/152db423e06f6d93a560b45b4330fac6507c7aa7/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReferenceHandler.cs#L16**

I thought that enum would be cleaner since allows us to illustrate the "none feature enabled" state.

Do you have a better way in mind to illustrate this?

Reply:
Got it. This way works fine.